packages/components/src/components/domainproperties/samples/SampleTypeDesigner.tsx

```diff
@@ -0,0 +1,394 @@
+import {SampleTypePropertiesPanel} from "./SampleTypePropertiesPanel";
+import {
+    Alert,
+    DomainDesign,
+    DomainDetails, generateId,
+    IDomainField,
+    KINDS, naturalSort,
+    resolveErrorMessage,
+    SAMPLE_TYPE,
+    saveDomain, SCHEMAS,
+    WizardNavButtons
+} from "../../..";
+import DomainForm from "../DomainForm";
+import {IParentOption} from "../../entities/models";
+import {IParentAlias} from "./models";
+import {addDomainField} from "../actions";
+import {initSampleSetSelects,} from "../../samples/actions";
+import {SAMPLE_SET_DISPLAY_TEXT, STICKY_HEADER_HEIGHT} from "../../../constants";
+import {fromJS, List, Map} from "immutable";
+
+const DEFAULT_SAMPLE_FIELD_CONFIG = {
+    required: true,
+    dataType: SAMPLE_TYPE,
+    conceptURI: SAMPLE_TYPE.conceptURI,
+    rangeURI: SAMPLE_TYPE.rangeURI,
+    lookupSchema: 'exp',
+    lookupQuery: 'Materials',
+    lookupType: {...SAMPLE_TYPE},
+    name: 'SampleId',
+} as Partial<IDomainField>;
+
+const NEW_SAMPLE_SET_OPTION: IParentOption = {
+    label: `(Current ${SAMPLE_SET_DISPLAY_TEXT})`,
+    value: "{{this_sample_set}}"
+} as IParentOption;
+
+export const SAMPLE_SET_IMPORT_PREFIX :string = 'materialInputs/';
+export const DATA_CLASS_IMPORT_PREFIX :string = 'dataInputs/';
+const DATA_CLASS_SCHEMA_KEY:string = 'exp/dataclasses';
+
+interface Props {
+    onCancel: () => void
+    onComplete: (response: any) => void
+    beforeFinish?: (formValues: {}) => void
+    initModel: DomainDetails
+    defaultSampleFieldConfig?: Partial<IDomainField>
+    includeDataClasses?: boolean
+
+    //EntityDetailsForm props
+    noun?: string
+    nameExpressionInfoUrl?: string
+    nameExpressionPlaceholder?: string
+
+    //DomainDesigner props
+    domainDesignerStickyHeight?: number
+}
+
+interface State {
+    model: SampleTypeModel
+    parentOptions: Array<IParentOption>
+    invalidDomainField: string
+
+    activePanelIndex: number
+
+    submitting: boolean
+    currentPanelIndex: number
+
+    error: React.ReactNode
+}
+
+export class SampleTypeDesigner extends React.PureComponent<Props, State> {
+    private _dirty = false;
+
+    static defaultProps = {
+        nameExpressionPlaceholder: 'S-\${now:date}-\${dailySampleCount}',
+        defaultSampleFieldConfig: DEFAULT_SAMPLE_FIELD_CONFIG,
+        noun: SAMPLE_SET_DISPLAY_TEXT,
+        domainDesignerStickyHeight: STICKY_HEADER_HEIGHT,
+        includeDataClasses: false,
+    };
+
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            submitting: false,
+            currentPanelIndex: 0,
+            model: undefined,
+            invalidDomainField: undefined,
+            activePanelIndex: 0,
+            parentOptions: undefined,
+            error: undefined,
+        };
+    }
+
+    componentDidMount = (): void => {
+        const domainDetails = this.props.initModel || DomainDetails.create();
+        const model = SampleTypeModel.create(domainDetails);
```

Comment:
Would there be any benefit to putting the above to lines up into the constructor so that the model state object is set there based on the initModel prop?

Reply:
moved.